lib/puppet/provider/esx_firewall_ruleset/default.rb

```diff
@@ -0,0 +1,101 @@
+provider_path = Pathname.new(__FILE__).parent.parent
+require File.join(provider_path, 'vcenter')
+require 'rbvmomi'
+
+Puppet::Type.type(:esx_firewall_ruleset).provide(
+  :default, 
+  :parent => Puppet::Provider::Vcenter
+) do
+
+
+  @doc = "Manages ESXi firewall rulesets"
+
+  def enable
+    firewall_config.EnableRuleset(:id => @resource[:name])
```

Comment:
@maniacmurphy If there is a bad ID (name) passed through, then this would get picked up from the ruleset method called from exists? before we even get here, if the name doesn't match a firewall ruleset you will get:

```ruby
raise Puppet::Error, "No firewall ruleset named #{name}" if set.empty?
```

If it's passed that method then we can be pretty sure the name is good and don't need to validate further IMO